nemo_skills/evaluation/top_answer_processor.py

```diff
@@ -0,0 +1,314 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import shutil
+import sys
+from collections import Counter, defaultdict
+from itertools import zip_longest
+from pathlib import Path
+from typing import Any, List, Tuple
+from enum import Enum
+
+import hydra
+from omegaconf import MISSING
+from tqdm import tqdm
+
+from nemo_skills.code_execution.math_grader import extract_answer
+from nemo_skills.evaluation.metrics import read_predictions
+from nemo_skills.utils import get_help_message, nested_dataclass, setup_logging, unroll_files
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class ProcessTopAnswerConfig:
+    """Top-level parameters for the script"""
+
+    # Input_dir relative to which all the input_files are specified
+    input_dir: str = MISSING
+    # Input files relative to input_dir which are used for majority voting
+    # Can specify multiple patterns separated by space
+    # e.g. "path/to/file1.jsonl path/to/file2.jsonl" or with regex
+    # "test_dir/output-rs*.jsonl"
+    input_files: Any = MISSING
+
+    # Output directory is optional depending on whether the task is to fill the majority answer
+    # or to just extract the best answer
+    output_dir: str | None = None
+
+    # The script can be run in two modes:
+    # 1. fill: use the best answer as the expected_answer to fill input_files
+    # 2. extract: identify the best answer from input_files
+    mode: str = MISSING
+
+    # where to put the majority answer. By default replacing the expected_answer (assuming it's unknown)
+    # but change to predicted_answer, to follow up with a judge evaluation
+    fill_key: str = "expected_answer"
+
+    # if True, will not change the fill_key if it's already filled with not None
+    ignore_if_not_none: bool = False
+
+    # if True, will use string match to fill is_correct key
+    fill_is_correct: bool = True
+
+    # if True, use the highest scoring answer from the RM as the majority answer
+    use_highest_rm_score: bool = False
+
+    # if True, will use the RM score for weighted majority voting
+    use_majority_rm_score: bool = False
+
+    # if provided, will fail if can't find this many files. Useful in scheduled
+    # pipelines to ensure this step doesn't run if some of the expected files are missing
+    require_num_files: int | None = None
+
+    def __post_init__(self):
+        """Building data_file from dataset/split if not provided directly."""
+        if isinstance(self.input_files, str):
+            self.input_files = self.input_files.split(" ")
```

Comment:
Let's also split on comma for consistency with other scripts, e.g. https://github.com/NVIDIA/NeMo-Skills/blob/main/nemo_skills/inference/retrieve_similar.py#L70